### PR TITLE
Gives a descriptive name to Interact signal printers and fixes a test with latest Reactive

### LINKED ISSUE #162: Fix Slider Signal string Test
Reactive changed the way signals are displayed, which broke this test: https://github.com/JuliaGizmos/Interact.jl/blob/master/test/ijulia.jl#L25

caught in: https://github.com/JuliaLang/METADATA.jl/pull/8931

When https://github.com/JuliaGizmos/Reactive.jl/pull/129 gets merged we'll fix the test.

### PR DESCRIPTION
**Pull request by JobJob**
Gives a descriptive name to Interact signal printer Signals and Fixes #162

### CODE REVIEW

**Comment by JobJob** [comment]
Changed the test to just include the `string(signal(sliderWidget))` in the test rather than hard coding. Now it doesn't fail when Reactive changes the way it `show`s signals.

Travis failed on Mac, 0.5, while all the other builds passed, not clear to me why, any ideas @tkelman ?

**Comment by tkelman** [comment]
It's been that way since https://travis-ci.org/JuliaGizmos/Interact.jl/builds/202528395 - someone needs to try to reproduce locally

**Comment by JobJob** [comment]
Managed to reproduce locally simply by running: `julia -e 'Pkg.test("Interact"; coverage=true)'` and managed to get the error to go away by removing the `coverage=true` - is that needed?

**Comment by tkelman** [comment]
It's needed if you want to measure coverage, but it's almost always a Julia bug if the value of that flag changes anything significant.

**Comment by JobJob** [comment]
Just looked at Reactive's travis.yml seems to measure coverage differently: https://github.com/JuliaGizmos/Reactive.jl/blob/master/.travis.yml

How to proceed?

**Comment by tkelman** [comment]
the submit step won't work if you haven't set the coverage flag to true, and if you don't list a script in .travis.yml then the default one has coverage enabled. try with 0.5.2 on the off chance that something I just backported fixed it, otherwise maybe do a reverse-bisect (new = works, old = broken) to find what was responsible for fixing it between 0.5 and 0.6

**Comment by JobJob** [comment]
ok cheers

**Comment by JobJob** [comment]
Preliminary results: doesn't work on 0.5.2, does work on 0.6.0-pre.alpha tag

**Comment by JobJob** [comment]
`d5521ebae8afd35c7a68e03e25375709d9f627fa` from Sep 8, 2016, is the first commit that runs the test ok. They break on the previous commit `340f31bc231f8c67307b17b3475886e14dfcc6c5`.

Many CPU hours died to bring us this information.

**Comment by tkelman** [comment]
Ah, so running Julia with `--inline=no` might also reproduce the issue even on Julia 0.6?

**Comment by JobJob** [comment]
No, this doesn't seem to have any errors with 0.6-rc1: `julia6 --inline=no -e 'Pkg.test("Interact"; coverage=true)'`

**Comment by KristofferC** [comment]
`Pkg.test()` starts a new process so presumably it will not run with `--inline=no` just because the parent process does?

**Comment by tkelman** [comment]
right, would need to do roughly `julia6 --inline=no test/runtests.jl`

**Comment by JobJob** [comment]
`julia runtests.jl` doesn't cause the error (even on 0.5), it only happens with `Pkg.test("Interact"; coverage=true)`  - is there another flag I should use? `--code-coverage=all` doesn't seem to trigger it

**Comment by JobJob** [comment]
Ok, starting julia with `--inline=no` and then `include("runtests.jl")` causes a `Segmentation fault: 11` on julia 0.5 and 0.6-rc1

What's next?

**Comment by tkelman** [comment]
Try with julia-debug and/or inside gdb or lldb to see if we can get a backtrace, report as Julia bug? If possible, trying to cut down the amount of code needed to trigger the segfault

**Comment by JobJob** [comment]
😬  may not get a chance to do that - little busy right now. I imagine this is reproducible by anyone with a mac.

**Comment by JobJob** [comment]
Ok I couldn't resist. Seems like it's happening here: https://github.com/JuliaLang/IJulia.jl/blob/master/src/heartbeat.jl#L16
called from https://github.com/JuliaGizmos/Interact.jl/blob/master/test/ijulia.jl#L12 -> https://github.com/JuliaLang/IJulia.jl/blob/master/src/init.jl#L89

**Comment by tkelman** [comment]
cc @stevengj do you have a mac you could check this on?

**Comment by JobJob** [comment]
Minimal reproduction:
```
using ZMQ

threadid = Vector{Int}(128) # sizeof(uv_thread_t) <= 8 on Linux, OSX, Win
sock = Socket(Context(), ROUTER)
function heartbeat_thread(sock::Ptr{Void})
    ccall((:zmq_device,ZMQ.zmq), Cint, (Cint, Ptr{Void}, Ptr{Void}),
          ZMQ.QUEUE, sock, sock)
    nothing
end
heartbeat_c = cfunction(heartbeat_thread, Void, (Ptr{Void},))
# @show heartbeat_c sock.data threadid
ccall(:uv_thread_create, Cint, (Ptr{Int}, Ptr{Void}, Ptr{Void}),
      threadid, heartbeat_c, sock.data)
```

**Comment by tkelman** [comment]
cc @vtjnash any ideas?

**Comment by vtjnash** [comment]
We may want to consider adding a type parameter to `Base.cconvert` to force specialization. With inlining disabled it's currently deciding it should box the return value to try to avoid generated too many specializations of this function:
```
SSAValue(0) = $(Expr(:invoke, MethodInstance for cconvert(::Type{T} where T, ::Int64), :(Base.cconvert), :(Main.Cint), 0))
```
In general, this is not solvable. On v0.6, you should be able to emit the `ccall` as a literal:
```
@eval function heartbeat_thread(sock::Ptr{Void})
    $(Expr(:foreigncall, :((:zmq_device, ZMQ.zmq), Int32,
        Core.svec(Int32, Ptr{Void}, Ptr{Void}), ZMQ.QUEUE, sock, sock))
    nothing
end
```
to help ensure that no extraneous code will get emitted.

**Comment by JobJob** [comment]
Ok, thanks. I've added osx julia 0.5 and removed julia nightly from the allowed failures.

File: .travis.yml [hunk 1]
```diff
@@ -10,7 +10,8 @@ notifications:
   email: false
 matrix:
   allow_failures:
-    - julia: nightly
+    - julia: 0.5
+      os: osx
```
Comment (tkelman) [inline .travis.yml]: might be better to just only enable coverage on linux

Comment (JobJob) [inline .travis.yml]: Good idea

**Comment by JobJob** [comment]
Actually in the end I just disabled code coverage for mac on 0.5 as suggested by tkelman. All looking pretty peachy now.

**Pull request merged**